lispify/lispify.py

```diff
@@ -57,63 +64,61 @@ def __init__(self, val):
         """
         Lispify a piece of data. Throws ValueError on failure.
         """
+        self.original_val = val
 
-        self.val = val
-
-        if self.should_parse():
+        if self.should_parse(val):
             self.val = self.parse_val(val)
         else:
-            raise ValueError("failed to lispify %s" % str(self.val))
+            raise ValueError(u'failed to lispify {}'.format(val))
 
-    def should_parse(self):
+    def should_parse(self, val):
         """
         If this returns False, LispType is invalid whatever the value.
         """
-
         return True
 
     def parse_val(self, val):
         """
         Override this to have special value manipulation.
         """
-
         return val
 
     def __repr__(self):
-        r = u"<%s object %s>" % (self.__class__.__name__,
-                                 self.__str__())
-        return r.encode('utf-8')
+        return u'{0}({1})'.format(self.__class__.__name__,
+                                  repr(self.original_val))
 
     def __str__(self):
-        return unicode(self).encode('utf-8')
+        if version_info.major >= 3:
```

Comment:
Have you considered http://python-future.org/compatible_idioms.html#custom-str-methods ?

Again, same considerations as for `basestring` above.